demo/code-demo/scripts/codesamples.js

```diff
@@ -0,0 +1,273 @@
+/*
+    This file contains the code samples which will appear live in the web-page.
+    Each sample method name starts with _Report_ or _Page or _Embed depends on which section it appears.
+    Please keep this.
+*/
+
+// ---- Embed Code ----------------------------------------------------
+
+function _Embed_BasicEmbed() {
+    var txtAccessToken = $('#txtAccessToken').val();
+    var txtEmbedUrl = $('#txtReportEmbed').val();
+    var txtEmbedReportId = $('#txtEmbedReportId').val();
+
+    var embedConfiguration = {
+        type: 'report',
+        accessToken: txtAccessToken,
+        embedUrl: txtEmbedUrl,
+        id: txtEmbedReportId,
+        settings: {
+            filterPaneEnabled: true,
+            navContentPaneEnabled: true
+        }
+    };
+    
+    var reportContainer = document.getElementById('reportContainer');
+    powerbi.embed(reportContainer, embedConfiguration);
+}
+
+function _Embed_EmbedWithDefaultFilter() {
+    var txtAccessToken = $('#txtAccessToken').val();
+    var txtEmbedUrl = $('#txtReportEmbed').val();
+    var txtEmbedReportId = $('#txtEmbedReportId').val();
+    
+    const filter = {
+      $schema: "http://powerbi.com/product/schema#basic",
+      target: {
+        table: "Store",
+        column: "Chain"
+      },
+      operator: "In",
+      values: ["Lindseys"]
+    };
+    
+    var embedConfiguration = {
+        type: 'report',
+        accessToken: txtAccessToken,
+        embedUrl: txtEmbedUrl,
+        id: txtEmbedReportId,
+        settings: {
+            filterPaneEnabled: false,
+            navContentPaneEnabled: false
+        },
+        filters: [filter]
+    };
+    
+    var reportContainer = document.getElementById('reportContainer');
+    powerbi.embed(reportContainer, embedConfiguration);
+}
+
+// ---- Report Operations ----------------------------------------------------
+
+ function _Report_GetId() {
+    report = powerbi.embeds[0];
+    $('#result').html(report.getId());
```

Comment:
But the code will not be that clear.
this is a sample code for developers, they should add the check after copying.